Components/TaskForm.js

```diff
@@ -1,93 +1,71 @@
+import Link from "next/link";
 import styled from "styled-components";
-
-const DEFAULT_VALUES = {
-  title: "",
-  description: "",
-  status: "Pending",
-};
+import ConfirmationDialog from "./ConfirmationDialog";
 
 export default function TaskForm({
-  onSubmit,
-  initiatives,
-  taskToEdit = null,
   isEditMode = false,
+  task,
+  initiativeId,
+  onSubmit,
 }) {
   const router = useRouter();
-  const { id: initiativeId } = router.query;
 
-  const selectedInitiative = initiatives.find(
-    (initiative) => initiative.id === initiativeId
-  );
-
-  const selectedTasksArray = selectedInitiative?.tasks ?? [];
   const [formData, setFormData] = useState({
-    ...DEFAULT_VALUES,
-    ...taskToEdit,
+    title: task?.title || "",
+    description: task?.description || "",
+    status: task?.status || "Pending",
   });
 
   const [errors, setErrors] = useState({});
   const [isDialogVisible, setIsDialogVisible] = useState(false);
 
-  function validateForm() {
-    const { title, description } = formData;
-    return {
-      title: title.trim() ? null : "Title is required",
-      description: description.trim() ? null : "Description is required",
-    };
-  }
-
   function handleChange(event) {
     const { name, value } = event.target;
     setErrors({});
     setFormData((prev) => ({ ...prev, [name]: value }));
   }
 
+  function validateForm() {
+    const { title, description } = formData;
+    return {
+      title: title.trim() ? undefined : "Title is required",
+      description: description.trim() ? undefined : "Description is required",
+    };
+  }
+
   function handleSubmit(event) {
-    event.preventDefault();
+    if (event) event.preventDefault();
     const newErrors = validateForm();
     setErrors(newErrors);
 
     if (Object.values(newErrors).some((error) => error)) {
       return;
     }
 
-    if (isEditMode) {
-      setIsDialogVisible(true);
-    } else {
-      saveChanges();
-    }
+    onSubmit(formData);
+    router.replace({
+      pathname: `/initiatives/${initiativeId}/tasks/${task?.id}`,
+      query: { success: "true" },
+    });
   }
 
-  function saveChanges() {
-    const updatedTask = {
-      id: taskToEdit?.id || crypto.randomUUID(),
-      ...formData,
-    };
-
-    const updatedTasks = isEditMode
-      ? selectedTasksArray.map((task) =>
-          task.id === updatedTask.id ? updatedTask : task
-        )
-      : [updatedTask, ...selectedTasksArray];
-
-    const updatedInitiative = {
-      ...selectedInitiative,
-      tasks: updatedTasks,
-    };
-
-    onSubmit(updatedInitiative);
+  function handleCancel(event) {
+    event.preventDefault();
+    setIsDialogVisible(true);
+  }
 
-    router.push({
-      pathname: `/initiatives/${initiativeId}/tasks/${updatedTask.id}`,
-      query: { success: "true" },
-    });
+  function navigateAway() {
+    setIsDialogVisible(false);
+    router.push(
+      isEditMode ? `/initiatives/${initiativeId}/tasks/${task?.id}` : "/"
```

Comment:
```suggestion
      isEditMode ? `/initiatives/${initiativeId}/tasks/${task._id}` : "/"
```

it is the key `_id` you want to use here.